src/utils/BackendClient/NimbledroidApiHandler.js

```diff
@@ -0,0 +1,103 @@
+import fetchJson from '../fetchJson';
+
+const matchSiteName = profileName => profileName
+  .replace(/.*.*http[s]*:\/\/[www]*\.*(.*)[/].*/, (match, firstMatch) => firstMatch);
+
+const scenarioShouldBeConsidered = ({ scenario_name, status }) => {
+  if (
+    status !== 'fail' &&
+    scenario_name.startsWith('customFlow') &&
+    scenario_name.includes('http')
+  ) {
+    return true;
+  }
+  return false;
+};
+
+const transformedDataForMetrisGraphics = (nimbledroidData) => {
```

Comment:
Similarly, I would just put this function inline in the one place where it is used, maybe with a quick comment like `// tranform data into format metrics graphics expects`

Reply:
IMHO I beg to disagree.
The current `fetchData` function is clear that it only does two things: it fetches and transforms the data. The reader does not need to look at 40 extra lines of code to determine if anything else is happening besides a transformation of the fetched data. The name of the function is self-explanatory.

I'm trying to follow some clean code pattern I've studying.
If you have no strong objections I will leave it as such.

Reply:
These questions don't have clear answers I suppose. I know some people prefer to have lots of little functions (even if they're only called once) as a means to decompose the structure of their program into some kind of logical hierarchy. Personally I don't find that helpful, I'd rather just see the entire control flow in one place for something like this since I can just read it from top to bottom and be done with it (you can find a longer argument on this here: https://medium.com/@copyconstruct/small-functions-considered-harmful-91035d316c29) . But I won't insist here if you feel strongly you want to code in this style.